### DISCUSSION TITLE
Which one is more gas efficient?

### DISCUSSION THREAD

**Discussion by Not-Sarthak** [Cyfrin/foundry-full-course-cu#2268] [Q&A] [answered]
Which one is more Gas Efficient?
- `abi.encodePacked()`
- `string.concat`

Also, how can I measure the benchmarks for other similar questions in order to try and learn out Gas Efficiency myself?

**Comment by EngrPips** [comment]
Hey @Not-Sarthak, I am not sure which one is more gas efficient as I haven't tried running them and comparing the gas consumption, but I assume you can do that yourself in Remix https://remix.ethereum.org/ and find out.

I am not sure what you mean by this `how can I measure the benchmarks for other similar questions in order to try and learn out Gas Efficiency myself?`

**Reply by Not-Sarthak** [reply]
I meant there are multiple ways to do the same thing but Option 1 might be more gas efficient than Option 2. In order to understand which is more gas efficient, will it come with the practise or are there any good blogs for it?

I hope i was able to explain properly this time 😅

**Reply by EngrPips** [reply, marked as answer]
Yeah, you did well to explain it. I mean, running the processes in Remix and comparing their gas consumption is one of the ways to know which of any two processes is more gas efficient. Honestly, The more deep you dive into `EVM` the more you just start to know naturally which process might be less gas-consuming.

**Reply by Not-Sarthak** [reply]
Yes got it! thank you :)

**Reply by EngrPips** [reply]
You are always welcome, Friend.
